content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -0,0 +1,186 @@
+
+```prisma
+model comments {
+   // Fields
+}
+```
+
+However, you can still follow the naming convention without renaming the underlying `comments` table in the database by using the [`@@map`](../../../reference/api-reference/prisma-schema-reference#map-1) <span class="api"></span> attribute:
+
+```prisma
+model Comment {
+   // Fields
+  @@map("comments")
+}
+```
+
+With this modified model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
+
+You can also [`@map`](../../../reference/api-reference/prisma-schema-reference#map) <span class="api"></span> a column/field name:
+
+```prisma highlight=2-4;normal
+model Comment {
+  content String  @map("comment_text")
+  email   String  @map("commenter_email")
+  type    Enum    @map("comment_type")
+
+  @@map("comments")
+}
+```
+
+You can also `@map` an enum value, or `@@map` an enum:
+
+```prisma highlight=3,5;normal
+enum Type {
+  Blog,
+  Twitter @map("comment_twitter")
+
+  @@map("comment_source_enum")
+}
+```
+
+`@map` and `@@map` allow you to [tune the shape of your Prisma Client API](../prisma-client/working-with-prismaclient/use-custom-model-and-field-names#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) by decoupling model and field names from table and column names in the underlying database.
+
+## Constraint and index names (Preview)
+
+In [2.29.0](https://github.com/prisma/prisma/releases/tag/2.29.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
+<Admonition type="warning">
+
+**Not supported by MongoDB**<br />
+Custom constraint and index names apply to relational databases only.
+
+</Admonition>
+
+### Enabling the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+Named constraints is a Preview feature. To enable this feature:
+
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
+
+2. Choose an [upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+<Admonition type="alert">
+
+If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#upgrade-paths-for-prisma-migrate-users).
```

Comment:
This also applies to other Migrate commands like `db push` I think, so needs to be formulated wider.

Reply:
Also: Is this the only case when the upgrade path is relevant for people to look at? This is pretty sepcific, but reading the upgrade guide itself makes it sound much more general than this admonition.

Reply:
Added push